Первый PR: Начало проекта

mixmaster1989 opened the pull request.

mixmaster1989 committed:
Создан и настроен каркас. GIT+GITFLOW+ENV+каркас кода

mixmaster1989 committed:
Создаем папку data и файл lessons

mixmaster1989 committed:
добавим уроков. теперь их 20

mixmaster1989 committed:
исправление ситаксиса в json

mixmaster1989 committed:
Markdown: Добавлено форматирование текста с использованием parse_mode: 'Markdown' для выделения заголовков, жирного текста и списков.
Эмодзи: Добавлены эмодзи для улучшения визуального восприятия.
Кнопки: Используются inline_keyboard для перехода к заданиям.

mixmaster1989 committed:
Кнопка "Начать обучение": Заменена команда /learn на кнопку 🚀 Начать обучение.
Ответы через кнопки: Вопросы теперь имеют 4 кнопки с вариантами ответа, где правильный вариант перемешивается с неправильными.
Перемешивание ответов: Добавлена функция shuffleAnswers для случайного порядка кнопок.
Переход к следующему уроку: После правильного ответа появляется кнопка ➡️ Следующий урок.

mixmaster1989 committed:
Поле wrongAnswers: В JSON добавлены неправильные ответы для каждого урока.
Сбор вариантов ответа: Варианты ответа собираются из правильного ответа и массива wrongAnswers.
Проверка ответа: Сравнение выбранного ответа с правильным.
Короткие ответы: Убедитесь, что ответы в JSON-файле достаточно короткие, чтобы помещаться в кнопки.

mixmaster1989 committed:
заполнение правильных и неправильных ответов

mixmaster1989 committed:
Теперь bot.js использует модуль storage.js для управления прогрессом, а функция safeSend централизует обработку ошибок.

mixmaster1989 committed:
module а не modules

mixmaster1989 committed:
Теперь бот поддерживает два уровня обучения:

Новичок — уроки из lessons.json.
Продвинутый — уроки из intermediate_lessons.json.

mixmaster1989 committed:
Добавлена кнопка "Я не новичок":
Эта кнопка устанавливает прогресс пользователя на начало уроков для продвинутого уровня.
Обновлена обработка callback_query:
Добавлена обработка start_intermediate, чтобы сразу начать уроки для продвинутого уровня.
Логика прогресса:
Для продвинутого уровня прогресс начинается с индекса, равного количеству уроков для новичков (beginnerLessons.length).
Теперь при нажатии на "Я не новичок" пользователь сразу начнёт с уроков для продвинутого уровня.

ThePositree commented:
```npm install``` работать не будет, потому что проект не инициализирован с помощью npm.
Нужно выполнить ```npm init```.

mixmaster1989 committed:
Теперь при нажатии на кнопку "Я не новичок" бот будет корректно выводить уроки для продвинутого уровня.

mixmaster1989 committed:
25 уроков для продвинутого уровня.

mixmaster1989 committed:
20 новых уроков для уровня "Новичок".
Темы включают console.log, операторы, преобразование типов, строгий режим, методы массивов, Math, Date, JSON, setTimeout, setInterval, try...catch и другие.
